Software/Development/SemesterProject/App/closedLoopControlManager.c

```diff
@@ -3,38 +3,56 @@
  *
  *  Created on: 25. apr. 2019
  *      Author: Nicolai Fransen
+ *
+ *      This module contains the closed loop control algorithm of the system
+ *
+ *      It collects all the different parts of the control system and calls
+ *      them in the correct order.
+ *
+ *      The flow is:
+ *          - Obtain references
+ *          - Obtain currents
+ *          - Obtain Rotor position angle in radians
+ *          - Transform currents from ABC -> dq
+ *          - Calculate dq voltage references from current PI-controllers
+ *          - Transform voltages from dq -> alpha/beta
+ *          - Run SVM with alpha/beta voltages as references.
  */
 
 #include "closedLoopControlManager.h"
 
 void runClosedLoopControl(void)
 {
     float movementReference = 0;
-    float iqReference = 0, idReference = 0;
+    float currentReferences[1];

```

Comment:
array definition in C array[numberOfElements], it is weird because then you can not access index numberOfElements, but that is how it is. Same for abcCurrents.

Did you test? Sometimes when a value outside of the array is accesed the system crashes very badly.